presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/TestMetastoreUtil.java

```diff
@@ -153,4 +167,34 @@ public void testHiveSchemaPartition()
         Properties actual = MetastoreUtil.getHiveSchema(ThriftMetastoreUtil.fromMetastoreApiPartition(TEST_PARTITION_WITH_UNSUPPORTED_FIELDS), ThriftMetastoreUtil.fromMetastoreApiTable(TEST_TABLE_WITH_UNSUPPORTED_FIELDS, TEST_SCHEMA));
         assertEquals(actual, expected);
     }
+
+    @Test
+    public void testComputePartitionKeyFilter()
```

Comment:
There are many layers between query and metastore communication and we already had regressions more than once. Can we also have an end-to-end test with Glue metastore showing the functionality works as intended?
This could go into `TestHiveGlueMetastore` to use real Glue. We probably should add GlueMetastoreStats metric to account for size of responses being pulled from the service.

Reply:
`TestHiveGlueMetastore` is doing low level tests of the metastore calls, so it wouldn't help with this issue. I filed #5709 to add product tests that work end-to-end.